unable to create connection on Windows 32 bit machine's with GO MODULES

I am unable to create the connection to the running redis when the go application is designed as GO MODULES on windows 32 bit computer. it works fine when being compiled and executed on 64 bit computer.

on 32 bit computer's golang module it's connect and immediately disconnect from redis.

Adding snap of the PANIC:

C:\Users\######\Documents\Switch\redis_test
λ go build && .\redis_test.exe
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x4022ac]

goroutine 1 [running]:
runtime/internal/atomic.Xadd64(0x1244c1e4, 0x1, 0x0, 0x5c77b4, 0xe)
        c:/go/src/runtime/internal/atomic/asm_386.s:105 +0xc
github.com/mediocregopher/radix/v3.(*Pool).newConn(0x1244c160, 0x5c77b4, 0xe, 0x0, 0x0, 0x0)
        D:/#################/pkg/mod/github.com/mediocregopher/radix/v3@v3.4.1/pool.go:433 +0x111
github.com/mediocregopher/radix/v3.NewPool(0x5c586e, 0x3, 0x5c75d8, 0xe, 0xa, 0x0, 0x0, 0x0, 0x12436030, 0x0, ...)
        D:/#################/pkg/mod/github.com/mediocregopher/radix/v3@v3.4.1/pool.go:322 +0x347
main.main()
        C:/Users/######/Documents/Switch/redis_test/main.go:11 +0x77

--- a/cluster.go
+++ b/cluster.go
@@ -110,12 +110,15 @@ func ClusterWithTrace(ct trace.ClusterTrace) ClusterOpt {
 // with it, including a set of pools to each of its instances. All methods on
 // Cluster are thread-safe
 type Cluster struct {
+	// Atomic fields must be at the beginning of the struct since they must be
+	// correctly aligned or else access may cause panics on 32-bit architectures
+	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	lastClusterdown int64 // unix timestamp in milliseconds, atomic
+
 	co clusterOpts
 
 	// used to deduplicate calls to sync
 	syncDedupe *dedupe
-
-	lastClusterdown int64 // unix timestamp in milliseconds, atomic
 
 	l              sync.RWMutex
 	pools          map[string]Client

--- a/pool.go
+++ b/pool.go
@@ -236,6 +236,11 @@ func PoolWithTrace(pt trace.PoolTrace) PoolOpt {
 // performance during low-concurrency usage. It can be disabled using
 // PoolPipelineWindow(0, 0).
 type Pool struct {
+	// Atomic fields must be at the beginning of the struct since they must be
+	// correctly aligned or else access may cause panics on 32-bit architectures
+	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	totalConns int64 // atomic, must only be access using functions from sync/atomic
+
 	opts          poolOpts
 	network, addr string
 	size          int
@@ -245,10 +250,6 @@ type Pool struct {
 	// when closed is true (closed is also protected by l)
 	pool   chan *ioErrConn
 	closed bool
-
-	// totalConns is an atomic value and should only be accessed by the atomic
-	// package.
-	totalConns int64
 
 	pipeliner *pipeliner
 

